File: Swift/stringcalculator/StringCalculator.swift
import UIKit

class StringCalculator{
    let default_result:Int = 0
    
    let numberExtractor:NumberExtractor
    let numberRemover:NumberRemover
    let numberValidator:NumberValidator
    
    init(numberExtractor:NumberExtractor,
         numberRemover:NumberRemover,
         numberValidator:NumberValidator){
        
       self.numberExtractor = numberExtractor
       self.numberRemover = numberRemover
       self.numberValidator = numberValidator
    }
    
    func add(input:String) throws -> Int {
        let allNumbers:[Int] = numberExtractor.extract(input);
        
        let numbersWithoutIgnored = numberRemover.removeNumbersToIgnore(allNumbers);
        
        try numberValidator.validate(numbersWithoutIgnored);
        
        return sumNumbers(numbersWithoutIgnored);
    }
    
        
    private func sumNumbers(numbers:[Int]) -> Int {
        var sum:Int = 0
    
        
        for number in numbers {
            sum += number
        }
        
        return sum;
    }
}

File: C#/StringCalculatorKata/StringCalculator.cs
using StringCalculatorKata.Boundaries;
using System;
using System.Collections.Generic;

namespace StringCalculatorKata
{
    public class StringCalculator
    {
        INumberExtractor numberExtractor;
        INumberRemover numberRemover;
        INumberValidator numberValidator;

        public StringCalculator(INumberExtractor numberExtractor,
                                INumberRemover numberRemover,
                                INumberValidator numberValidator)
        {
            this.numberExtractor = numberExtractor;
            this.numberRemover = numberRemover;
            this.numberValidator = numberValidator;
        }

        public int Add(String input)
        {
            List<int> allNumbers = numberExtractor.Extract(input);

            List<int> numbersWithoutIgnored = numberRemover.RemoveNumbersToIgnore(allNumbers);

            numberValidator.Validate(numbersWithoutIgnored);

            return SumNumbers(numbersWithoutIgnored);
        }

        private int SumNumbers(List<int> numbers)
        {
            int sum = 0;
            foreach (int num in numbers)
            {
                sum += num;
            }
            return sum;
        }
    }
}
